specs/_features/maxeb_increase/capella.py

```diff
@@ -1627,6 +1652,83 @@ def process_pending_balance_deposits(state: BeaconState) -> None:
     state.pending_balance_deposits = state.pending_balance_deposits[next_pending_deposit_index:]
 
 
+def apply_pending_consolidation(state: BeaconState, pending_consolidation: PendingConsolidation) -> None:
+    current_epoch = get_current_epoch(state)
+
+    # Resolve the target of consolidation
+    target_index = resolve_consolidated_to(state, pending_consolidation.target_index)
+    target_validator = state.validators[target_index]
+    source_validator = state.validators[pending_consolidation.source_index]
+
+    # Apply consolidation
+    source_validator.consolidated_to = target_index
+
+    # Don't move the balance and slash the target if the source was slashed
+    if is_slashed(source_validator):
+        if is_slashed_proposer(source_validator):
+            penalty = PROPOSER_EQUIVOCATION_PENALTY_FACTOR * EFFECTIVE_BALANCE_INCREMENT
+            decrease_balance(state, target_validator, penalty)
+            initiate_validator_exit(state, target_index)
+            target_validator.slashed = add_flag(target_validator.slashed, SLASHED_PROPOSER_FLAG_INDEX)
+
+        if is_slashed_attester(source_validator):
+            if is_attester_slashable_validator(target_validator, current_epoch):
+                slash_validator(state, target_index)
+
+        return
+
+    # Don't move the balance and slash the source if the target was slashed
+    if is_slashed(target_validator):
+        if is_slashed_proposer(target_validator):
+            penalty = PROPOSER_EQUIVOCATION_PENALTY_FACTOR * EFFECTIVE_BALANCE_INCREMENT
+            decrease_balance(state, source_validator, penalty)
+            initiate_validator_exit(state, pending_consolidation.source_index)
+            source_validator.slashed = add_flag(source_validator.slashed, SLASHED_PROPOSER_FLAG_INDEX)
+
+        if is_slashed_attester(target_validator):
+            if is_attester_slashable_validator(source_validator, current_epoch):
+                slash_validator(state, pending_consolidation.source_index)
+
+        return
+
+    # Don't move the balance if the source withdrew
+    if source_validator.withdrawable_epoch < current_epoch:
+        return
```

Comment:
why do we need this?

Reply:
In the very edge case at the time of processing consolidation the source active balance could be withdrawn and topped-up after that so the balance is > 0. This check is to avoid activation of a balance that was never active bypassing the churn.

Although this is very much an edge case because it requires more than 24 hours of non finality, the check makes sense to me.